test/widgets/action_sheet_test.dart

```diff
@@ -99,46 +107,244 @@ void main() {
+                'emoji_name': emoji.emojiName,
+              });
+        });
+
+        testWidgets('${emoji.emojiName} removing success', (tester) async {
+          final message = eg.streamMessage(
+            reactions: [Reaction(
+              emojiName: emoji.emojiName,
+              emojiCode: emoji.emojiCode,
+              reactionType: ReactionType.unicodeEmoji,
+              userId: eg.selfAccount.userId)]
+          );
+          await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+          connection.prepare(json: {});
+          await tapButton(tester);
+          await tester.pump(Duration.zero);
+
+          check(connection.lastRequest).isA<http.Request>()
+            ..method.equals('DELETE')
+            ..url.path.equals('/api/v1/messages/${message.id}/reactions')
+            ..bodyFields.deepEquals({
+                'reaction_type': 'unicode_emoji',
+                'emoji_code': emoji.emojiCode,
+                'emoji_name': emoji.emojiName,
+              });
+        });
+
+        testWidgets('${emoji.emojiName} request has an error', (tester) async {
+          final message = eg.streamMessage();
+          await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+          connection.prepare(httpStatus: 400, json: {
+            'code': 'BAD_REQUEST',
+            'msg': 'Invalid message(s)',
+            'result': 'error',
           });
+          await tapButton(tester);
+          await tester.pump(Duration.zero); // error arrives; error dialog shows
+
+          await tester.tap(find.byWidget(checkErrorDialog(tester,
+            expectedTitle: 'Adding reaction failed',
+            expectedMessage: 'Invalid message(s)')));
+        });
+      }
     });
 
-    testWidgets('request has an error', (tester) async {
-      final message = eg.streamMessage();
-      await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+    group('emoji picker;', () {
+      Future<void> setupEmojiPicker(WidgetTester tester, {
+        required Message message,
+        required Narrow narrow,
+      }) async {
+        final httpClient = FakeImageHttpClient();
+        debugNetworkImageHttpClientProvider = () => httpClient;
+        httpClient.request.response
+          ..statusCode = HttpStatus.ok
+          ..content = kSolidBlueAvatar;
+
+        await setupToMessageActionSheet(tester, message: message, narrow: narrow);
+        store.setServerEmojiData(ServerEmojiData(codeToNames: {
+          '1f4a4': ['zzz', 'sleepy'], // (just 'zzz' in real data)
+        }));
+        await store.handleEvent(RealmEmojiUpdateEvent(id: 1, realmEmoji: {
+          '1': eg.realmEmojiItem(emojiCode: '1', emojiName: 'buzzing'),
+        }));
+
+        await tester.tap(find.ancestor(
+          of: find.byIcon(ZulipIcons.chevron_right),
+          matching: find.byType(InkWell)));
```

Comment:
nit: simplify and put more into user-visible terms:
```suggestion
        await tester.tap(find.byIcon(ZulipIcons.chevron_right));
```